lighthouse-core/lib/dependency-graph/simulator/connection-pool.js

```diff
@@ -93,6 +93,21 @@ module.exports = class ConnectionPool {
     }
   }
 
+  /**
+   * @param {Array<TcpConnection>} connections
+   */
+  _findConnectionWithLargestCongestionWindow(connections) {
+    /** @type {TcpConnection|null} */
+    let maxConnection = null;
+    for (let i = 0; i < connections.length; i++) {
+      const connection = connections[i];
+      const currentMax = (maxConnection && maxConnection.congestionWindow) || -Infinity;
```

Comment:
could remove this branching for `currentMax` by starting `maxConnection` at the first entry in `connections`, and starting the loop at `i = 1`. 

I assume you wouldn't need to bother with a `.length > 0` check because the previous code didn't bother.

Reply:
I think no matter what I'll need a `ts-ignore` or a check on `maxConnection`. I don't think ts is smart enough to know that starting the loop at `1` means that `maxConnection` is necessarily defined

Reply:
the explicit `return null` was a good call, very nice